### ISSUE TITLE
indentation_linter reports incorrect indentation when closing paren is not isolated

### ISSUE THREAD

**Issue by aronatkins** [r-lib/lintr#2011]
```r
safe_install_impl <- function(pkg_name) {}
safe_install <- function(pkg_name) {
  tryCatch(
    safe_install_impl(pkg_name),
    error = function(e) {
      e$message <- paste0(
        "Failed to install package ", pkg_name, ": ", e$message)
      stop(e)
    })
  cat("installed ", pkg_name, "\n")
}
```

This code reports:

```
stuff.R:4:4: style: [indentation_linter] Hanging indent should be 11 spaces but is 4 spaces.
    safe_install_impl(pkg_name),
   ^~~~~~~~
stuff.R:7:8: style: [indentation_linter] Hanging indent should be 26 spaces but is 8 spaces.
        "Failed to install package ", pkg_name, ": ", e$message)
       ^~~~~~~~~~~~~~~~~~~
stuff.R:8:6: style: [indentation_linter] Indentation should be 13 spaces but is 6 spaces.
      stop(e)
     ^~~~~~~~
```

If it is modified to have isolated closing parens, it does not produce warnings:

```r
safe_install_impl <- function(pkg_name) {}
safe_install <- function(pkg_name) {
  tryCatch(
    safe_install_impl(pkg_name),
    error = function(e) {
      e$message <- paste0(
        "Failed to install package ", pkg_name, ": ", e$message
      )
      stop(e)
    }
  )
  cat("installed ", pkg_name, "\n")
}
```

Is this an intentional change? The hanging indent message does not indicate that the closing paren is what triggers an indentation expectation for the code to align with the opening paren.

Note: The second code sample was produced with styler, so this is not a consistency problem.

**Comment by AshesITR** [comment]
Hi, this looks intentional.
Having a token on the same line as closing parenthesis triggers a so-called hanging indent, which is expected to be intended to the opening parenthesis, like so:

```
foo(a = 2,
    b = 3)
```

**Comment by jeksterslab** [comment]
I am not sure if this is relevant to the issue but here is my experience with the latest release. I use `styler` (1.10.1) and then `lintr` (3.1.0) to style R code first before linting. In the toy example below, the `original` code gets `styled with styler` like below. The styled code, however, gets flagged by the latest release of `lintr` with a similar error code as the OP.

`original`

```r
foo <- function(pkg = c("styler", "lintr", "usethis",
  "tidyverse", "ggplot2", "rmarkdown", "knitr")) {
  print(pkg)
}
```

`styled with styler`

```r
foo <- function(pkg = c(
                  "styler", "lintr", "usethis",
                  "tidyverse", "ggplot2", "rmarkdown", "knitr"
                )) {
  print(pkg)
}
```

**Comment by AshesITR** [comment]
Can you file your example as a separate issue, @jeksterslab?
Your case seems to be a more clear-cut false positive.

lintr 3.1.0 expects indentation to be

```r
foo <- function(pkg = c(
  "styler", "lintr", "usethis",
  "tidyverse", "ggplot2", "rmarkdown", "knitr"
)) {
  print(pkg)
}
```

which looks wonky to me.

**Comment by AshesITR** [comment]
You could set `hanging_indent_style = "never"` if you want to configure indentation to always use block indents.
